OasisDEX/oasis-borrow
# Add current position state to manage stETH position

# [Add current position state to manage stETH position](https://app.shortcut.com/oazo-apps/story/6052/add-current-position-state-to-manage-steth-position)

Includes changes for [Add banner to indicate user has interacted with other assets on AAVE with their DS-Proxy](https://app.shortcut.com/oazo-apps/story/6053/add-banner-to-indicate-user-has-interacted-with-other-assets-on-aave-with-their-ds-proxy)

## Changes 👷‍♀️
 - added lending pool calls, `getUserConfiguration` for reading user assets in one call and `getReservesList` to properly utilize the data from user configuration
 - added methods to parse above data
 - added a banner to an view (manage/adjust) position with other assets info
 - added a blocking error on the opening flow, on setting multiply (first step with proxy address)
 - added displaying proper position data (except out of scope) on view (manage/adjust)
  
## How to test 🧪
 - go to `/earn/open/aave-steth` with connected wallet which proxy has more assets than ETH/STETH on AAVE, you shouldn't be able to proceed beyond setting multiply
 - go to `/earn/aave-steth/<aave_proxy_address>`, if you have more assets than ETH/STETH on AAVE there should be a banner informing you that it's not a whole position
 - in above situation the left block should display proper position data (except out of scope fields)
 
 ## Images 🎨 
 
![image](https://user-images.githubusercontent.com/5095527/195380491-604c3448-7da7-42bb-969b-a491dd511ecf.png)
Regular view position

![image](https://user-images.githubusercontent.com/5095527/195380711-d5ea3326-fd06-4500-bc4b-17aff23b94d5.png)
View position with other assets on AAVE

![image](https://user-images.githubusercontent.com/5095527/195381062-5c7a8797-b5b5-4b1a-bafc-162aad902c97.png)
Open position with fields locked (can't go further, other assets on AAVE)
